datadog_checks_dev/datadog_checks/dev/tooling/commands/meta_commands/prometheus.py

```diff
@@ -0,0 +1,279 @@
+
+    if num_metrics:
+        echo_success('Number of metrics: {}'.format(num_metrics))
+    else:
+        echo_warning('No metrics!')
+        return
+
+    if num_gauge:
+        echo_info('Type `gauge`: {}'.format(num_gauge))
+
+    if num_counter:
+        echo_info('Type `counter`: {}'.format(num_counter))
+
+    if num_histogram:
+        echo_info('Type `histogram`: {}'.format(num_histogram))
+
+
+@prom.command(
+    context_settings=CONTEXT_SETTINGS,
+    short_help='Interactively parse metric info from a Prometheus endpoint'
+)
+@click.argument('endpoint')
+@click.argument('check')
+@click.option('--here', '-x', is_flag=True, help='Output to the current location')
+@click.pass_context
+def parse(ctx, endpoint, check, here):
+    """Interactively parse metric info from a Prometheus endpoint."""
+    if here:
+        output_dir = os.getcwd()
+    else:
+        output_dir = path_join(get_root(), check)
+        if not dir_exists(output_dir):
+            abort(
+                'Check `{check}` does not exist; try `ddev create{repo_flag} {check}`.'.format(
+                    check=check,
+                    repo_flag=' -e' if ctx.obj['repo_choice'] == 'extras' else ''
+                )
+            )
+
+    endpoint = sanitize_endpoint(endpoint)
+
+    echo_waiting('Scraping `{}`...'.format(endpoint))
+    metrics = parse_metrics(endpoint)
+    num_metrics = len(metrics)
+
+    echo_success('\nGlobally available options:')
+    echo_info('    t - Append .total to the available options')
+    echo_info('    s - Skip')
+    echo_info('    q - Quit')
+
+    for i, (metric, data) in enumerate(sorted(iteritems(metrics)), 1):
+        metric_parts = metric.split('_')
+        metric_template = '{}'.join(metric_parts)
+        num_separators = len(metric_parts) - 1
+
+        metric_options = [
+            metric_template.format(*possible_separators)
+            for possible_separators in product(METRIC_SEPARATORS, repeat=num_separators)
+        ]
+        num_options = len(metric_options)
+
+        default_option = num_options
+        options_prompt = 'Choose an option (default {}, as-is): '.format(default_option)
+        options_text = get_options_text(metric_options)
+
+        finished = False
+        choice_error = ''
+        progress_status = '({} of {}) '.format(i, num_metrics)
+        indent = ' ' * len(progress_status)
+
+        while not finished:
+            echo_success('\n{}{}'.format(progress_status, metric))
+
+            echo_success('Type: ', nl=False, indent=indent)
+            echo_info(data.get('type', 'None'))
+
+            echo_success('Info: ', nl=False, indent=indent)
+            echo_info(data.get('description', 'None'))
+
+            echo_info(options_text)
+
+            if choice_error:
+                echo_warning(choice_error)
+
+            echo_waiting(options_prompt, nl=False)
+
+            if num_options >= 9:
+                choice = ensure_unicode(input())
+            else:
```

Comment:
Should we make it so we always sanitize input? (perhaps this would be good as some util function?)

Reply:
No click does it for us